### PR TITLE
session: cache global variable for a short while

### PR DESCRIPTION
**Pull request by tiancaiamao**
When there are a lot of connections come to TiDB simultaneously, load
global variable would make TiKV meta region overload.
We can cache global variable for a short while to avoid this problem.

@coocood  @shenli

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -1282,14 +1289,37 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
...
-		vars.CommonGlobalLoaded = false
-		log.Errorf("Failed to load common global variables.")
-		return errors.Trace(err)
+
+	var rows []types.Row
+	var fields []*ast.ResultField
+	var err error
+
+	now := time.Now()
+	globalVariableCache.RLock()
```
Comment (ngaut) [inline session/session.go]: Extract to a function or method.

**Comment by shenli** [comment]
Should we update the cache when updating the value of some variables?

**Comment by jackysp** [comment]
I think it needs a test for the cache out of date.

File: session/session.go [hunk 2]
```diff
@@ -1272,6 +1272,34 @@ const loadCommonGlobalVarsSQL = "select HIGH_PRIORITY * from mysql.global_variab
...
+	gvc.lastModify = time.Now()
+	gvc.rows = rows
+	gvc.fields = fields
+	gvc.Unlock()
+}
+
+func (gvc *globalVariableCache) Get() (succ bool, rows []types.Row, fields []*ast.ResultField) {
+	gvc.RLock()
+	defer gvc.RUnlock()
+	if time.Now().Sub(gvc.lastModify) < 2*time.Second {
```
Comment (coocood) [inline session/session.go]: Make it constant.
And I think 2 seconds is too small, how about 10 seconds?

Comment (tiancaiamao) [reply session/session.go]: 10 seconds is too long!
2 seconds maybe a good balance between correctness and performance.

**Comment by tiancaiamao** [comment]
@jackysp I **have to** update some test, otherwise CI would fail.
I think that covers this change already.

**Comment by tiancaiamao** [comment]
@shenli  Update the cache doesn't bring significant advantages, I think.

**Pull request merged**